src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -224,34 +223,21 @@ public sealed class Column
             /// </summary>
             public DataViewType ColumnType { get; set; }
 
-            /// <summary>
-            /// Whether the column is a computed type.
-            /// </summary>
-            public bool IsComputed { get { return Generator != null; } }
-
             /// <summary>
             /// The generator function. if the column is computed.
             /// </summary>
-            public Delegate Generator { get; set; }
+            internal Delegate Generator { get; set; }
 
-            public Type ReturnType => Generator?.GetMethodInfo().GetParameters().LastOrDefault().ParameterType.GetElementType();
+            internal Type ReturnType => Generator?.GetMethodInfo().GetParameters().LastOrDefault().ParameterType.GetElementType();
 
-            public Column(IExceptionContext ectx, string memberName, DataViewType columnType,
-                string columnName = null, IEnumerable<AnnotationInfo> annotationInfos = null, Delegate generator = null)
+            public Column(string memberName, DataViewType columnType,
```

Comment:
>public [](start = 12, length = 6)

I would like to make this internal as well. Is there any scenario where users would want to add new columns to the `SchemaDefinition` generated from the type?

Reply:
I think so. In fact I think that's one of the major scenarios, isn't it? You start with an empty schema definition, then you add columns to it to describe how you want fields mapped to columns, etc.? This object exists in large part for those situations where you don't or somehow can't rely purely on the reflection based mechanism.

---
In reply to: [266437352](https://github.com/dotnet/machinelearning/pull/2995#discussion_r266437352) [](ancestors = 266437352)

Reply:
I was thinking you start with a schema that's auto-generated from the type, then modify the columns that are already in it. The schema generated from the type contains all the possible columns, why would we need to add new columns (as opposed to modifying existing columns)?

---
In reply to: [266605420](https://github.com/dotnet/machinelearning/pull/2995#discussion_r266605420) [](ancestors = 266605420,266437352)